src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldLookupMultiselect.class.php

```diff
@@ -142,22 +142,20 @@ public function FetchMLTRecords()
     protected function GetMLTFilterQuery()
     {
         $foreignTableName = $this->GetForeignTableName();
-
-        $oTableConf = new TCMSTableConf();
-        $oTableConf->LoadFromField('name', $foreignTableName);
-        $sNameField = $oTableConf->GetNameColumn();
-
-        $query = 'SELECT * FROM `'.MySqlLegacySupport::getInstance()->real_escape_string($foreignTableName).'` AS parenttable
-    WHERE 1=1
-    '.$this->GetMLTRecordRestrictions().'';
-        $bShowCustomsort = $this->oDefinition->GetFieldtypeConfigKey('bAllowCustomSortOrder');
-        if (true == $bShowCustomsort) {
-            $query .= 'ORDER BY MLT.`entry_sort` ASC , `parenttable`.`'.MySqlLegacySupport::getInstance()->real_escape_string($sNameField).'`';
-        } else {
-            $query .= 'ORDER BY `parenttable`.`'.MySqlLegacySupport::getInstance()->real_escape_string($sNameField).'`';
-        }
-
-        return $query;
+        $foreignTableConf = new TCMSTableConf();
+        $foreignTableConf->LoadFromField('name', $foreignTableName);
+        $foreignFieldName = $foreignTableConf->GetNameColumn();
+
+        $foreignTableNameQuoted = $this->getDatabaseConnection()->quoteIdentifier($foreignTableName);
+        $foreignFieldNameQuoted = $this->getDatabaseConnection()->quoteIdentifier($foreignFieldName);
```

Comment:
Please fetch services only once within one method and save the result in a variable.